src/nvim/spell.c

```diff
@@ -4292,23 +4290,22 @@ static void suggest_trie_walk(suginfo_T *su, langp_T *lp, char_u *fword, bool so
                     && utf_iscomposing(utf_ptr2char(fword
                                                     + sp->ts_fcharstart))) {
                   sp->ts_score -= SCORE_SUBST - SCORE_SUBCOMP;
-                } else if (!soundfold
-                           && slang->sl_has_map
+                } else if (!soundfold && slang->sl_has_map
                            && similar_chars(slang,
-                                            mb_ptr2char(tword
-                                                        + sp->ts_twordlen
-                                                        - sp->ts_tcharlen),
-                                            mb_ptr2char(fword
-                                                        + sp->ts_fcharstart))) {
+                                            utf_ptr2char(tword + sp->ts_twordlen
+                                                         - sp->ts_tcharlen),
+                                            utf_ptr2char(fword
+                                                         + sp->ts_fcharstart))
+                           ) {
```

Comment:
You should not move `)` on its own line. As far as I understand style guide you need to resolve the situation by either moving `!soundfold` (the very *first* part of condition) on its own line and indent it with +4 spaces or move `slang` (the very *first* argument) on its own line with +4 spaces. Following parts of condition or arguments should be aligned with the first one (like now).

Reply:
Ok, i'll go for moving `!soundfold` to a new line. If all fails, there is a possibility of refactoring out the expression `tword + sp->ts_twordlen - sp->ts_tcharlen` as it repeats in neighboring conditions

Reply:
Here's what can be done following what you siggest @ZyX-I. Only problem is that the rightmost brace reaches column 81 and the linter complains only about that.
Is there another possiblity for fitting this code?
```
                } else if (!soundfold
                           && slang->sl_has_map
                           && similar_chars(slang,
                                            utf_ptr2char(tword
                                                         + sp->ts_twordlen
                                                         - sp->ts_tcharlen),
                                            utf_ptr2char(fword
                                                         + sp->ts_fcharstart))) {
                  // For a similar character adjust score from
                  // SCORE_SUBST to SCORE_SIMILAR.
                  sp->ts_score -= SCORE_SUBST - SCORE_SIMILAR;
                }
```

Reply:
@ZviRackover It is not what I suggested and I do not see `!soundfold` on a new line there. What I suggested is

```C
                } else if (
                    !soundfold
                    && slang->sl_has_map
                    && similar_chars(slang,
                                     utf_ptr2char(tword
                                                  + sp->ts_twordlen
                                                  - sp->ts_tcharlen),
                                     utf_ptr2char(fword
                                                  + sp->ts_fcharstart))) {
                  // For a similar character adjust score from
                  // SCORE_SUBST to SCORE_SIMILAR.
                  sp->ts_score -= SCORE_SUBST - SCORE_SIMILAR;
                }
```

Same thing may be done with `slang` as well, I think doing both would be cleaner:

```C
                } else if (
                    !soundfold
                    && slang->sl_has_map
                    && similar_chars(
                        slang,
                        utf_ptr2char(tword + sp->ts_twordlen - sp->ts_tcharlen),
                        utf_ptr2char(fword + sp->ts_fcharstart))) {
                  // For a similar character adjust score from
                  // SCORE_SUBST to SCORE_SIMILAR.
                  sp->ts_score -= SCORE_SUBST - SCORE_SIMILAR;
                }
```

Reply:
Thanks. Now i see what you mean.